First draft of Types proposal

erget opened the pull request:
Here's a first draft of a proposal on using netCDF-4 types in CF2 files. I'd like to hear what the community has to say about it.

erget committed:
First draft of Types proposal

czender reviewed (changes requested):
Thank you for working on this, Daniel. I think it needs more background and work before inviting the CF list as a whole to examine it. In particular:
1. Please move this proposal to a separate directory, e.g., "newtypes" or "enum" etc., on the same level as "group" for this proposal. 
2. Note that all atomic external integer types are already covered by this proposal (that I wrote) https://cf-trac.llnl.gov/trac/ticket/166
That proposal is for CF1. It does not include NC_STRING, though allowing NC_STRING in CF1 has been discussed on the list before. Please familiarize yourself with the status of that.
3. The changes above will help to avoid repeating many discussions that have already been had. Once that is done, I'll be happy to approve this PR and I think it will lead to more productive discussions.

erget committed:
Incorporate discussions from Trac 166

Thanks for the feedback - I've tried to incorporate it into this newer commit.
Concerning the individual concerns:

1. This was already done in the previous commit, may have been obfuscated by
the fact that I also edited the README to point to the new proposal. This has
made a conflict which I'll resolve in the next commit.
2. I remember seeing this in the agenda for last year's workshop but somehow
the ticket had slipped my mind. I've incorporated the content (in my opinion)
into the proposal, the only things that I've removed where retained for
backwards compatibility. This probably won't be possible for CF-2 so I didn't
attempt it, but if you don't agree I'm happy to talk about it.
3. I've also explicitly referenced the ticket in the proposal so that we have a
trace back to the original document. It may still be worth pursuing using
unsigned types in CF1 separately, but but it made more sense to me to highlight
this as a separate one.

erget committed:
Merge branch 'master' into enums

czender reviewed (approved):
Thank you for addressing previous comments. I'll merge this then add my suggestions as a new PR in a day or two.
